### PR TITLE
[Checkbox] Remove NDL fallbacks

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-ux/issues/596

### WHAT is this pull request doing?

Removes references to newDesignLanguage in the Checkbox component

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Review by alex-page — APPROVED** [review]
This looks great, thanks @aveline ⚡️

**Review by emmaorhun — APPROVED** [review]

File: src/components/Checkbox/Checkbox.scss [hunk 1]
```diff
@@ -124,28 +63,22 @@
...
-  }
-
-  // We need to know if the input is focusing to change the border color.
-  .Input:focus + .Backdrop,
-  .Input:active + .Backdrop {
-    @include control-backdrop(focused-error);
+// stylelint-disable selector-max-specificity, selector-max-class, selector-max-combinators, max-nesting-depth
+.error {
+  .Icon {
+    @include recolor-icon(var(--p-icon-on-critical, color('red', 'dark')));
```
Comment (kyledurand) [inline src/components/Checkbox/Checkbox.scss:79]:
```suggestion
    @include recolor-icon(var(--p-icon-on-critical));
```

**Review by kyledurand — COMMENTED** [review]
This is looking good but there are a couple of fallbacks that can still be removed. The one I've commented on and the one on line 59

````
@include recolor-icon(var(--p-icon-on-interactive, color('indigo')));
```

**Comment by aveline** [comment]
Thanks for catching those @kyledurand!

**Review by kyledurand — APPROVED** [review]
👏

**Pull request merged**